tests/test_line_profiler.py

```diff
@@ -16,113 +13,76 @@ def g(x):
-        self.assertEqual(lp.code_map, {f.__code__: {}})
-        lp = LineProfiler(f, g)
-        self.assertEqual(lp.functions, [f, g])
-        self.assertEqual(lp.code_map, {
-            f.__code__: {},
-            g.__code__: {},
-        })
-
-    def test_enable_disable(self):
-        lp = LineProfiler()
-        self.assertEqual(lp.enable_count, 0)
-        lp.enable_by_count()
-        self.assertEqual(lp.enable_count, 1)
-        lp.enable_by_count()
-        self.assertEqual(lp.enable_count, 2)
-        lp.disable_by_count()
-        self.assertEqual(lp.enable_count, 1)
-        lp.disable_by_count()
-        self.assertEqual(lp.enable_count, 0)
-        self.assertEqual(lp.last_time, {})
-        lp.disable_by_count()
-        self.assertEqual(lp.enable_count, 0)
+def test_init():
+    lp = LineProfiler()
+    assert lp.functions == []
+    assert lp.code_map == {}
+    lp = LineProfiler(f)
+    assert lp.functions == [f]
+    assert lp.code_map == {f.__code__: {}}
+    lp = LineProfiler(f, g)
+    assert lp.functions == [f, g]
+    assert lp.code_map ==  {
+        f.__code__: {},
+        g.__code__: {},
+    }
+
+
+def test_enable_disable():
+    lp = LineProfiler()
+    assert lp.enable_count == 0
+    lp.enable_by_count()
+    assert lp.enable_count == 1
+    lp.enable_by_count()
+    assert lp.enable_count == 2
+    lp.disable_by_count()
+    assert lp.enable_count == 1
+    lp.disable_by_count()
+    assert lp.enable_count == 0
+    assert lp.last_time == {}
+    lp.disable_by_count()
+    assert lp.enable_count == 0
+
+    with lp:
+        assert lp.enable_count == 1
+        with lp:
+            assert lp.enable_count == 2
+        assert lp.enable_count == 1
+    assert lp.enable_count == 0
+    assert lp.last_time == {}
 
+    with pytest.raises(RuntimeError):
+        assert lp.enable_count == 0
         with lp:
-            self.assertEqual(lp.enable_count, 1)
-            with lp:
-                self.assertEqual(lp.enable_count, 2)
-            self.assertEqual(lp.enable_count, 1)
-        self.assertEqual(lp.enable_count, 0)
-        self.assertEqual(lp.last_time, {})
-
-        with self.assertRaises(RuntimeError):
-            self.assertEqual(lp.enable_count, 0)
-            with lp:
-                self.assertEqual(lp.enable_count, 1)
-                raise RuntimeError()
-        self.assertEqual(lp.enable_count, 0)
-        self.assertEqual(lp.last_time, {})
-
-    def test_function_decorator(self):
-        profile = LineProfiler()
-        f_wrapped = profile(f)
-        self.assertEqual(f_wrapped.__name__, 'f')
-
-        self.assertEqual(profile.enable_count, 0)
-        value = f_wrapped(10)
-        self.assertEqual(profile.enable_count, 0)
-        self.assertEqual(value, f(10))
-
-        timings = profile.code_map[f.__code__]
-        self.assertEqual(len(timings), 2)
-        for timing in timings.values():
-            self.assertEqual(timing.nhits, 1)
-
-    def test_gen_decorator(self):
```

Comment:
I thought I had a patch that fixed and re-enabled all of these tests. Maybe it got lost?

Reply:
That's possible. I don't recall the patch though, so I have no clue where to find it.